### PR TITLE
Added form-wide onChange functionality

### LINKED ISSUE #883: v6 form onChange
I would like to listen if the form changes some value. Then submit myself the form at every change (there are not text inputs but checkboxes etc.) I look at the easiest solution.. but I figured out that I have to listen at Field component level on changes, wrap the field props onChange and execute the handleSubmit method after it executes the fields onChange method one.

Maybe we can introduce a onChange method in the reduxForm configuration?

So #806 this issue could be solved too with this

### PR DESCRIPTION
**Pull request by erikras**
Was suggested and attempted in #2530.

Fixes #883.

### CODE REVIEW

**Comment by fzaninotto** [comment]
Where is this fantastic new feature documented?

**Comment by gustavohenke** [comment]
Uh, nowhere, at the moment!

**Comment by jscinoz** [comment]
This has some odd interaction with initialValues when the new global onChange event is used to submit the form on change.

In the reduxForm decorator's mapStateToProps, there is the following:

```javascript
const shouldUpdateInitialValues = enableReinitialize && initialized && !deepEqual(initialValues, stateInitial)
const shouldResetValues = shouldUpdateInitialValues && !keepDirtyOnReinitialize

let initial = initialValues || stateInitial || empty

if (shouldUpdateInitialValues) {
  initial = stateInitial || empty
}

let values = getIn(formState, 'values') || initial

if (shouldResetValues) {
  values = initial
}
```

Specifically:


```javascript
if (shouldUpdateInitialValues) {
  initial = stateInitial || empty
}
```

This always uses the state values, which are stale as initialize() is not called until reduxForm's componentWillReceiveProps (i.e. the component rendered by the connect() HOC). The end result is an infinite update loop (and the tab eventually being killed by the browser) as values & initialValues will continuously alternate.

File: src/reduxForm.js [hunk 1]
```diff
@@ -235,6 +235,11 @@ const createReduxForm =
             this.validateIfNeeded(nextProps)
             this.warnIfNeeded(nextProps)
             this.submitIfNeeded(nextProps)
+            if (nextProps.onChange) {
```
Comment (egeste) [inline src/reduxForm.js]: 👍

**Comment by timhwang21** [comment]
@erikras just wondering if there is a minor version release planned soon. This patch in particular would be really useful to me. As usual, thanks for the great library!

**Comment by erikras** [comment]
Published in [`v6.6.0`](https://github.com/erikras/redux-form/releases/tag/v6.6.0).

**Comment by j-machado** [comment]
I'm having the same issue as @jscinoz in 6.6.0, can't use `enableReinitialize` with `onChange`. 

For now, I'm using an ugly flag to skip unwanted calls to `onChange`. Did anyone had a better solution for this?

**Comment by ravicious** [comment]
@jscinoz @j-machado: I think it may be related to #2427 and #2153.

**Pull request merged**